js/woo.js

```diff
@@ -0,0 +1,1465 @@
+const Precise = require ('./base/Precise');
+const { TICK_SIZE } = require ('./base/functions/number');
+
+// ---------------------------------------------------------------------------
+
+module.exports = class woo extends Exchange {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'id': 'woo',
+            'name': 'Woo',
+            'countries': [ 'KY' ], // Cayman Islands
+            'rateLimit': 1000, // No defaults known
+            'version': 'v1',
+            'certified': false,
+            'hostname': 'woo.org',
+            'has': {
+                'createOrder': true,
+                'cancelOrder': true,
+                'cancelOrders': true,
+                'cancelAllOrders': undefined,
+                'createMarketOrder': undefined,
+                'fetchBalance': true,
+                'fetchCanceledOrders': undefined,
+                'fetchClosedOrder': undefined,
+                'fetchClosedOrders': undefined,
+                'fetchCurrencies': true,
+                'fetchDepositAddress': true,
+                'fetchDeposits': true,
+                'fetchFundingRateHistory': undefined,
+                'fetchMarkets': true,
+                'fetchMyTrades': true,
+                'fetchOHLCV': true,
+                'fetchOrder': true,
+                'fetchOrderBook': true,
+                'fetchOrders': true,
+                'fetchOrderTrades': true,
+                'fetchOpenOrder': undefined,
+                'fetchOpenOrders': undefined,
+                'fetchPosition': undefined,
+                'fetchPositions': undefined,
+                'fetchStatus': undefined,
+                'fetchTicker': undefined,
+                'fetchTickers': undefined,
+                'fetchTime': undefined,
+                'fetchTrades': true,
+                'fetchTransfers': true,
+                'fetchTransactions': true,
+                'fetchWithdrawals': true,
+                'deposit': undefined,
+                'withdraw': undefined,
+                'addMargin': undefined,
+                'reduceMargin': undefined,
+                'setLeverage': undefined,
+            },
+            'timeframes': {
+                '1m': '1m',
+                '5m': '5m',
+                '15m': '15m',
+                '30m': '30m',
+                '1h': '1h',
+                '4h': '4h',
+                '12h': '12h',
+                '1d': '1d',
+                '1w': '1w',
+                '1M': '1mon',
+                '1y': '1y',
+            },
+            'urls': {
+                'logo': '  <<<TODO>>>    ',
+                'api': {
+                    'public': 'https://api.woo.org',
+                    'private': 'https://api.woo.org',
+                    // TODO: official api doesnt mention, but there seems to be :  API_TESTNET_URL = "http://api.staging.woo.network" as said https://github.com/wanth1997/python-wootrade/blob/main/wootrade/client.py
```

Comment:
Let's add that under `test` – see how that is done with the other exchanges. Upon adding a testnet, let's make sure that it works – test the testnet.

Reply:
Let's add these test urls, or lmk if I am missing something.